### PR TITLE
chore(data-warehouse): refactor logic

### PR DESCRIPTION
**Pull request by EDsCODE**
## Problem

- logics in the editor component are messy

## Changes

- consolidate in a new logic

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+                    path: model.path.split('/').pop(),
+                }
+            })
+            localStorage.setItem(editorModelsStateKey(props.key), JSON.stringify(queries))
+        },
+        runQuery: ({ queryOverride }) => {
+            actions.setActiveQuery(queryOverride || values.queryInput)
+        },
+        saveAsView: async () => {
+            LemonDialog.openForm({
```
Comment (Gilbert09) [inline frontend/src/scenes/data-warehouse/editor/multitabEditorLogic.tsx:253]: Oh wild, didnt know you could do this from a logic

File: frontend/src/scenes/data-warehouse/settings/source/SourceConfiguration.tsx [hunk 2]
```diff
@@ -43,7 +43,7 @@ function UpdateSourceConnectionFormContainer(props: UpdateSourceConnectionFormCo
         <>
             <span className="block mb-2">Overwrite your existing configuration here</span>
             <Form logic={dataWarehouseSourceSettingsLogic} formKey="sourceConfig" enableFormOnSubmit>
-                <SourceFormComponent {...props} jobInputs={source?.job_inputs} />
```
Comment (Gilbert09) [inline frontend/src/scenes/data-warehouse/settings/source/SourceConfiguration.tsx:46]: Is this intentional?

**Review by Gilbert09 — APPROVED** [review]
Looks good, not tested it, but trust that it works and is behind a flag etc - thanks for sorting this out!

File: frontend/src/scenes/data-warehouse/settings/source/SourceConfiguration.tsx (hunk 2 above)
Comment (EDsCODE) [reply frontend/src/scenes/data-warehouse/settings/source/SourceConfiguration.tsx:46]: nope 😬

**Pull request merged**